calendar-bundle/contao/classes/Calendar.php

```diff
@@ -141,13 +141,15 @@ protected function generateFiles($arrFeed)
 		$time = time();
 
 		// Get the upcoming events
-		$objArticle = CalendarEventsModel::findUpcomingByPids($arrCalendars, $arrFeed['maxItems']);
+		$objArticles = CalendarEventsModel::findUpcomingByPids($arrCalendars, $arrFeed['maxItems']);
 
 		// Parse the items
-		if ($objArticle !== null)
+		if ($objArticles !== null)
 		{
-			while ($objArticle->next())
+			while ($objArticles->next())
 			{
+				$objArticle = $objArticles->current();
```

Comment:
If `$objArticles` is not used inside the loop, we can switch to a foreach in these cases I think.
```suggestion
			foreach ($objArticles as $objArticle)
			{
```

Reply:
Yes, I know. Using `current()` was less of a hassle though. 😄